### PR TITLE
OASIS-8551 fix(attribute-validation): Adds validation on user attributes to only allow primitive types.

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Summary
- Adds validation on user attributes to only allow primitive types.
- Adds unit tests.

## Test plan
All tests should pass

# ISSUE
OASIS-8551

### CODE REVIEW

**Review by Mat001 — APPROVED** [review]
Looks good to me. 
The valid types are those that are supported by the data backend, make sure that is aligned.

**Review by jaeopt — APPROVED** [review]
LGTM

File: pkg/event/factory.go [hunk 1]
```diff
@@ -291,3 +291,23 @@ func getTagValue(eventTags map[string]interface{}) (float64, error) {
 
 	return 0, errors.New("no event tag found for value")
 }
+
+// check if attribute value is valid
+func isValidAttribute(value interface{}) bool {
```
Comment (msohailhussain) [inline pkg/event/factory.go:296]: can be in switch case.

**Review by msohailhussain — APPROVED** [review]
lgtm

**Pull request merged**